pkg/kubelet/pod_workers.go

```diff
@@ -138,57 +135,61 @@ type podWorkers struct {
 func newPodWorkers(syncPodFn syncPodFnType, recorder record.EventRecorder, workQueue queue.WorkQueue,
 	resyncInterval, backOffPeriod time.Duration, podCache kubecontainer.Cache) *podWorkers {
 	return &podWorkers{
-		podUpdates:                map[types.UID]chan UpdatePodOptions{},
-		isWorking:                 map[types.UID]bool{},
-		lastUndeliveredWorkUpdate: map[types.UID]UpdatePodOptions{},
-		syncPodFn:                 syncPodFn,
-		recorder:                  recorder,
-		workQueue:                 workQueue,
-		resyncInterval:            resyncInterval,
-		backOffPeriod:             backOffPeriod,
-		podCache:                  podCache,
+		podUpdates:     map[types.UID]chan UpdatePodOptions{},
+		isWorking:      map[types.UID]bool{},
+		syncPodFn:      syncPodFn,
+		recorder:       recorder,
+		workQueue:      workQueue,
+		resyncInterval: resyncInterval,
+		backOffPeriod:  backOffPeriod,
+		podCache:       podCache,
 	}
 }
 
-func (p *podWorkers) managePodLoop(podUpdates <-chan UpdatePodOptions) {
+func (p *podWorkers) managePodLoop(uid types.UID, podUpdates <-chan UpdatePodOptions) {
 	var lastSyncTime time.Time
-	for update := range podUpdates {
-		err := func() error {
-			podUID := update.Pod.UID
-			// This is a blocking call that would return only if the cache
-			// has an entry for the pod that is newer than minRuntimeCache
-			// Time. This ensures the worker doesn't start syncing until
-			// after the cache is at least newer than the finished time of
-			// the previous sync.
-			status, err := p.podCache.GetNewerThan(podUID, lastSyncTime)
-			if err != nil {
+
+	for {
+		select {
+		case update := <-podUpdates:
```

Comment:
Unlike using `range` over the channel, this will continue to loop when the channel is closed, because reads from a closed channel just return the zero value. Is that really what you want?

Reply:
Agree. I should check whether the channel is closed.I will make an update later